### PR TITLE
Settings refactoring & other small issues

### PR DESCRIPTION
**Pull request by math07**
WIP

### CODE REVIEW

File: forms/mainwindow.ui [hunk 1]
```diff
@@ -940,19 +950,12 @@
...
            <layout class="QVBoxLayout" name="verticalLayout_10">
             <item>
-             <widget class="QPushButton" name="pushButton_ResetSettings">
-              <property name="text">
-               <string>Reset Settings back to saved state</string>
-              </property>
-             </widget>
-            </item>
-            <item>
              <widget class="QPushButton" name="pushButton_FillSettingsWithDefaults">
```
Comment (shinnok) [inline forms/mainwindow.ui]: I don't think we need this button anymore, for 3 settings.

File: johnny.pro [hunk 2]
```diff
@@ -9,6 +9,12 @@ greaterThan(QT_MAJOR_VERSION, 4): QT += widgets
 
 TARGET = johnny
 
+# The application version
+VERSION = 2.0
```
Comment (shinnok) [inline johnny.pro]: This is a good addition. Thanks.

File: src/johnsession.cpp [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+        m_limitShells = m_settings.value("limitShells").toString();
+    }
+    if (m_settings.contains("limitSalts")) {
+        m_limitSalts = m_settings.value("limitSalts").toInt();
+    }
+    // Advanced options
+    if (m_settings.contains("nbForkProcess")) {
+        m_nbProcess = m_settings.value("nbForkProcess").toInt();
+    }
+    m_ui->spinBox_nbOfOpenMPThread->setValue(m_settings.value("OMP_NUM_THREADS").toInt());
```
Comment (shinnok) [inline src/johnsession.cpp]: Where's is this m_ui coming from? JohnSession shouldn't know about it. If it's work in progress ignore.

File: src/johnsession.h [hunk 4]
```diff
@@ -0,0 +1,88 @@
+#ifndef JOHNSESSION_H
+#define JOHNSESSION_H
+
+#include <QString>
+#include <QSettings>
+
+class JohnSession
```
Comment (shinnok) [inline src/johnsession.h]: I'm pretty much contempt with what you did in regards to JohnSession. Good job!

**Comment by math07** [comment]
I finished implementation yesterday night. I tested and manually reviewed the refactored code this morning. I think that we can merge unless you have new comments.

**Pull request merged**